presto-main/src/main/java/com/facebook/presto/sql/analyzer/ExpressionAnalyzer.java

```diff
@@ -400,7 +400,7 @@ private Type handleResolvedField(Expression node, FieldId fieldId, Type resolved
                 }
             }
 
-            Field field = baseScope.getRelationType().getFieldByIndex(fieldId.getFieldIndex());
```

Comment:
I think there is better approach. You can change signature of this method to:
```
handleResolvedField(Expression node, FieldId fieldId, Field field, StackableAstVisitorContext<Context> context)
```
and then modify callers as:
```
private Type handleResolvedField(Expression node, ResolvedField resolvedField, StackableAstVisitorContext<Context> context)
{
    return handleResolvedField(node, FieldId.from(resolvedField), resolvedField.getField(), context);
}
```
and
```
public Type visitFieldReference(FieldReference node, StackableAstVisitorContext<Context> context)
{
    Field field = baseScope.getRelationType().getFieldByIndex(node.getFieldIndex());
    return handleResolvedField(node, new FieldId(baseScope.getRelationId(), node.getFieldIndex()), field, context);
}
```

Reply:
Thanks.  I did a variation.